enclave-tls/src/tls_wrappers/openssl/negotiate.c

```diff
@@ -43,8 +44,10 @@ tls_wrapper_err_t openssl_internal_negotiate(tls_wrapper_ctx_t *ctx, unsigned lo
 		return -TLS_WRAPPER_ERR_NO_MEM;
 
 	X509_STORE *cert_store = SSL_CTX_get_cert_store(ssl_ctx->sctx);
-	X509_STORE_set_ex_data(cert_store, 0, ctx);
-	ETLS_DEBUG("openssl_internal_negothiate 's tls_ctx is %x\n", ctx);
+	int etls_ex_data = X509_STORE_get_ex_new_index(0, "etls_ex_data", NULL, NULL, NULL);
+	X509_STORE_set_ex_data(cert_store, etls_ex_data, ctx);
+
+	pthread_setspecific(g_key, (void *)&etls_ex_data);
```

Comment:
etls_ex_data is a variable stored on stack. So you cannot save its address as TLS.

Reply:
it's an address exactly.  Need store the address to key

Reply:
It is obviously wrong so please fix it. No matter why it is working now.